Exercise-master/src/main/java/org/upgrad/controllers/UserController.java

```diff
@@ -0,0 +1,133 @@
+        String userPresent = String.valueOf(userService.findUserByEmail(email));
+        if ( ! (userPresent.equalsIgnoreCase("null") )) {
+            return true ;
+        } else  {
+            return false ;
+        }
+    }
+
+
+    /*
+     It checks signup the user along with it's details if user is not already present.
+     This method gives relevant messages in case username or email is already registered.
+     */
+
+    @PostMapping("/api/user/signup/")
+    public ResponseEntity registerUser(@RequestParam String userName, @RequestParam String email, @RequestParam String password , @RequestParam String firstName, @RequestParam String lastName, @RequestParam String aboutMe, @RequestParam String contactNumber, @RequestParam String dob, @RequestParam String country) throws Exception{
+
+        // Setting url data into user model class.
+        User user = new User();
+        user.setPassword(hashPassword(password));
+        user.setLastName(lastName);
+        user.setFirstName(firstName);
+        user.setEmail(email);
+        user.setCountry(country);
+        user.setContactNumber(contactNumber);
+        user.setUserName(userName);
+        user.setDob(new SimpleDateFormat("dd/MM/yyyy").parse(dob));
+        user.setAboutMe(aboutMe);
+
+        String message = null ;
+
+       if ( registerUserWithExistingUsername(user.getUserName()) )
+       {
+            message = "Try any other Username, this Username has already been taken." ;
+            return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }  else if  ( registerUserWithExistingEmail(user.getEmail())) {
+           message = "This user has already been registered, try with any other emailId." ;
+           return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }  else {
+
+             //TODO : Make Parameters optional
+              userService.registerUserDetails(user);
+              message = userName + " successfully registered" ;
+              return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }
+    }
+
+
+    @PostMapping("/api/user/login")
+    public ResponseEntity loginWithBadCredentials(@RequestParam String userName, @RequestParam String password)
+    {
+        String message =  null ;
+        String passwordHash = hashPassword(password) ;
+
+       String passwordU =  String.valueOf(userService.findUserPassword(userName));
+       if(! (passwordU.equalsIgnoreCase(passwordHash)))
+       {
+           message = "Invalid Credentials"  ;
+       }
+       else {
+          String role =  String.valueOf(userService.findUserRole(userName));
+
+          if (role.equalsIgnoreCase("admin"))
+          {
+              // TODO :  Session Object Login
+              message = "You have logged in as admin!"  ;
+          } else if (role.equalsIgnoreCase("user"))
+          {
+              message = "You have logged in successfully!"  ;
+          }
+       }
+       return new ResponseEntity<>( message , HttpStatus.OK) ;
```

Comment:
Here we are only changing message, so in case credentials are invalid, HTTPStatus will be "OK" which shouldnt happen.
In case of invalid credentials return this:
return new ResponseEntity<>("Invalid Credentials", HttpStatus.UNAUTHORIZED);